src/mlpack/methods/ann/dists/normal_distribution_impl.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file normal_distribution_impl.hpp
+ * @author xiaohong ji
+ * @author Nishant Kumar
+ *
+ * Implementation of the Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "normal_distribution.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution()
+{
+  // Nothing to do here.
+}
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution(
+    const DataType& mean,
+    const DataType& sigma) :
+    mean(mean),
+    sigma(sigma)
+{
+}
+
+template<typename DataType>
+DataType NormalDistribution<DataType>::Sample() const
+{
+  return sigma * arma::randn<DataType>(mean.n_elem) + mean;
+}
+
+template<typename DataType>
+DataType NormalDistribution<DataType>::LogProbability(
+    const DataType& observation) const
+{
+  const DataType variance = arma::square(sigma);
+  DataType v1 = arma::log(sigma) + std::log(std::sqrt(2 * M_PI));
+  DataType v2 = arma::square(observation - mean) / (2 * variance);
```

Comment:
I guess since `variance` is only used once we can inline the calculation here as well: `arma::square(observation - mean) / (2 * arma::square(sigma));`.